### ISSUE TITLE
[config] [dynamic-config] Map configurations inconsistency between nodes

### ISSUE THREAD

**Issue by aarnaout** [hazelcast/hazelcast#10860]
This is similar but not exactly the same of the following:
https://github.com/hazelcast/hazelcast/issues/851
https://github.com/hazelcast/hazelcast/issues/592

I am using the following code to pre-configure a few maps when running Hazelcast instance:
```
Iterator hzIterator = Hazelcast.getAllHazelcastInstances().iterator();
HazelcastInstance hzinsance = null;
while (hzIterator.hasNext()) {
            hzinsance = (HazelcastInstance) hzIterator.next();
            Config config = hzinsance.getConfig();
            MapConfig mapConfig = config.getMapConfig("default");
            mapConfig.setMaxSizeConfig(new MaxSizeConfig(defaultMaxSize , MaxSizeConfig.MaxSizePolicy.PER_NODE)) 
                    .setEvictionPolicy(EvictionPolicy.LRU)  //Last Recently Used
                    .setTimeToLiveSeconds(defaultTtl);
}
```

When I run my two nodes, everything is good on both of them. However, when one node gets restarted, the configurations go back to default despite the code above! Yet in mancenter still showing the correct configurations from the other node. Once that other node with correct configurations gets restarted, then all the nodes have the wrong configurations, and that will be reflected in Mancenter when clicking Map Config

@jerrinot any idea? I know you were working on https://github.com/hazelcast/hazelcast/issues/10814 which could be related

**Comment by mmedenjak** [comment]
@aarnaout yes, you are right, currently there is no support for dynamically adding config. There is a new "dynamic config" feature coming up in 3.9. The configuration should be copied to newly joining nodes. Can you take a look and see if it fits your use case?

**Comment by jerrinot** [comment]
@aarnaout: can you try this with 3.9 EA? you will need to slighly change your code, something like this should do:
```java
Config config = instance.getConfig();
MapConfig mapConfig = new MapConfig("myMap");
mapConfig.setFoo().setBar();
config.addMapConfig(mapConfig);
```
this should work on 3.9 EA. It's OK to call it on just a single cluster instance - cluster will automatically distribute it too all members. Including the ones joining later. 

Caveat: You can use this to submit new configurations, but you cannot change existing configuration. At least for now.

**Comment by ahmetmircik** [comment]
Hi @aarnaout, any update on this?

**Comment by aarnaout** [comment]
So there were two issues indeed. 

1) Setting up the "default" map configurations the way I had it above was not working for the second node after it restarts. I changed that to within the bean xml configs, and that took care of this part:
<hz:map name="default" eviction-policy="LRU" max-size="1000" time-to-live-seconds="10000"></hz:map>
This works fine in both 3.8 & 3.9-EA

2) The other maps that I have specific eviction configurations for (different than default) were going back to default in this scenario:
Two node cluster, node 2 was restarted at some point, then a while after node 1 stops. All the maps that were copied to node 2 after it restarted picked up the "default" configurations instead of their specific configs. 
This issues seems to be resolved in **3.9-EA** (using code provided by jerrinot), so all good now.

**Comment by vbekiaris** [comment]
Thanks for the heads up @aarnaout , closing this issue now.
